src/roiextractors/extractors/tiffimagingextractors/scanimagetiff_utils.py

```diff
@@ -86,6 +86,111 @@ def parse_matlab_vector(matlab_vector: str) -> list:
+
+    Notes
+    -----
+    The ScanImage TIFF format includes:
+    1. TIFF Header Section: Defines byte order and offsets
+    2. ScanImage Static Metadata Section: Contains metadata applicable to all frames
+        - Non-Varying Frame Data: System configuration for all frames
+        - ROI Group Data: Defined regions of interest (if available)
+    3. Frame Sections: One per image frame, containing:
+        - IFD Header: Image File Directory with tags and values
+        - Frame-specific data: Timestamps and other frame metadata
+        - Image data: The actual image pixels
+
+    The function uses the tifffile module's read_scanimage_metadata function to extract
+    the raw metadata, then processes it to standardize key imaging parameters.
+    """
+    from tifffile import read_scanimage_metadata
+
+    with open(file_path, "rb") as fh:
+        all_metadata = read_scanimage_metadata(fh)
+        non_varying_frame_metadata = all_metadata[0]
+        roi_group_metadata = all_metadata[1]
+
+    if non_varying_frame_metadata["SI.hStackManager.enable"]:
+        num_slices = non_varying_frame_metadata["SI.hStackManager.numSlices"]
+
+        flyback_frames = (
+            non_varying_frame_metadata["SI.hStackManager.numFramesPerVolumeWithFlyback"]
+            - non_varying_frame_metadata["SI.hStackManager.numFramesPerVolume"]
+        )
+        # num_planes = num_slices + flyback_frames   # TODO: discuss on issue, leave the current behavior now
+        num_planes = num_slices
+        frames_per_slice = non_varying_frame_metadata["SI.hStackManager.framesPerSlice"]
+    else:
+        num_planes = 1
+        frames_per_slice = 1
+
+    if num_planes == 1:
+        sampling_frequency = non_varying_frame_metadata["SI.hRoiManager.scanFrameRate"]
+    else:
+        sampling_frequency = non_varying_frame_metadata["SI.hRoiManager.scanVolumeRate"]
+
+    # `channelSave` indicates whether the channel is saved. Note that a channel might not be saved even if it is active.
+    # We check `channelSave` first but keep the `channelsActive` check for backward compatibility.
+    channel_availability_keys = ["SI.hChannels.channelSave", "SI.hChannels.channelsActive"]
+    for channel_availability in channel_availability_keys:
+        if channel_availability in non_varying_frame_metadata.keys():
+            break
+
+    available_channels = non_varying_frame_metadata[channel_availability]
+    available_channels = [available_channels] if not isinstance(available_channels, list) else available_channels
+    channel_indices = np.array(available_channels) - 1  # Account for MATLAB indexing
+    channel_names = non_varying_frame_metadata["SI.hChannels.channelName"]
+    channel_names_available = [channel_names[i] for i in channel_indices]
+    num_channels = len(channel_names_available)
+    if roi_group_metadata:
+        roi_metadata = roi_group_metadata["RoiGroups"]
+    else:
+        roi_metadata = None
+
+    metadata_parsed = dict(
```

Comment:
I am adding this for backwards compatibility but probably not strictly necessary.